python/vsi/tools/time_utils.py

```diff
@@ -0,0 +1,101 @@
+import functools
+from time import time
+
+
+def format_time_string(seconds):
+  """
+  Format a time string the way I want it.
+  You would think I wouldn't have to write this myself!
```

Comment:
There are general packages that do this, e.g., [arrow](https://arrow.readthedocs.io/en/latest/), [pendulum](https://github.com/sdispater/pendulum), but for a one-off this is fine.